rust/processor/src/db/common/models/default_models/parquet_move_resources.rs

```diff
@@ -0,0 +1,153 @@
+        write_set_change_index: i64,
+        txn_version: i64,
+        block_height: i64,
+        block_timestamp: chrono::NaiveDateTime,
+    ) -> Self {
+        let parsed_data = Self::convert_move_struct_tag(
+            write_resource
+                .r#type
+                .as_ref()
+                .expect("MoveStructTag Not Exists."),
+        );
+        Self {
+            txn_version,
+            block_height,
+            write_set_change_index,
+            resource_type: write_resource.type_str.clone(),
+            fun: parsed_data.fun.clone(),
+            resource_address: standardize_address(&write_resource.address.to_string()),
+            module: parsed_data.module.clone(),
+            generic_type_params: parsed_data.generic_type_params,
+            data: Some(write_resource.data.clone()),
+            is_deleted: false,
+            state_key_hash: standardize_address(
+                hex::encode(write_resource.state_key_hash.as_slice()).as_str(),
+            ),
+            block_timestamp,
+        }
+    }
+
+    pub fn from_delete_resource(
+        delete_resource: &DeleteResource,
+        write_set_change_index: i64,
+        txn_version: i64,
+        block_height: i64,
+        block_timestamp: chrono::NaiveDateTime,
+    ) -> Self {
+        let parsed_data = Self::convert_move_struct_tag(
+            delete_resource
+                .r#type
+                .as_ref()
+                .expect("MoveStructTag Not Exists."),
+        );
+        Self {
+            txn_version,
+            block_height,
+            write_set_change_index,
+            resource_type: delete_resource.type_str.clone(),
+            fun: parsed_data.fun.clone(),
+            resource_address: standardize_address(&delete_resource.address.to_string()),
+            module: parsed_data.module.clone(),
+            generic_type_params: parsed_data.generic_type_params,
+            data: None,
+            is_deleted: true,
+            state_key_hash: standardize_address(
+                hex::encode(delete_resource.state_key_hash.as_slice()).as_str(),
+            ),
+            block_timestamp,
+        }
+    }
+
+    pub fn convert_move_struct_tag(struct_tag: &MoveStructTagPB) -> MoveStructTag {
+        MoveStructTag {
+            resource_address: standardize_address(struct_tag.address.as_str()),
+            module: struct_tag.module.to_string(),
+            fun: struct_tag.name.to_string(),
+            generic_type_params: struct_tag
+                .generic_type_params
+                .iter()
+                .map(|move_type| -> Result<Option<String>> {
+                    Ok(Some(
+                        serde_json::to_string(move_type).context("Failed to parse move type")?,
+                    ))
+                })
+                .collect::<Result<Option<String>>>()
+                .unwrap_or(None),
+        }
+    }
+
+    pub fn get_outer_type_from_resource(write_resource: &WriteResource) -> String {
```

Comment:
This method and `convert_move_struct_tag` don't have anything to do with `MoveResource`. Can you move them to the relevant struct or just make them freestanding?

As a general rule, if a method doesn't take `self` as a param or return `Self` there is no reason for it to a method on a struct.

Reply:
synced offline, will follow up in the next pr.